testing/functional_tests_assertions.rst

```diff
@@ -13,16 +13,15 @@ When doing functional tests, sometimes you need to make complex assertions in
 order to check whether the ``Request``, the ``Response`` or the ``Crawler``
 contain the expected information to make your test succeed.
 
-Here is an example with plain PHPUnit::
+The following example uses plain PHPUnit to assert that the response redirects
+to a certain URL::
 
-    $this->assertGreaterThan(
-        0,
-        $crawler->filter('html:contains("Hello World")')->count()
-    );
+    $this->assertSame(301, $client->getResponse()->getStatusCode());
```

Comment:
Maybe a bit off topic but I learned that one should use self::assertSame() over $this->assertSame()

Can you give more insights about this @localheinz? Thank you 🙏

Reply:
Let's merge this and we can change it later if needed. Thanks.